tree.go

```diff
@@ -418,32 +429,41 @@ walk: // Outer loop for walking the tree
 				idxc := path[0]
 				for i, c := range []byte(n.indices) {
 					if c == idxc {
-						if strings.HasPrefix(n.children[len(n.children)-1].path, ":") {
-							skipped = &skip{
-								path: prefix + path,
-								paramNode: &node{
-									path:      n.path,
-									wildChild: n.wildChild,
-									nType:     n.nType,
-									priority:  n.priority,
-									children:  n.children,
-									handlers:  n.handlers,
-									fullPath:  n.fullPath,
-								},
+						//  strings.HasPrefix(n.children[len(n.children)-1].path, ":") == n.wildChild
+						if n.wildChild {
+							skippedPath = prefix + path
+							latestNode = &node{
+								path:      n.path,
+								wildChild: n.wildChild,
+								nType:     n.nType,
+								priority:  n.priority,
+								children:  n.children,
+								handlers:  n.handlers,
+								fullPath:  n.fullPath,
 							}
 						}
 
 						n = n.children[i]
+
+						// match '/', If this condition is matched, the next route is found
+						if len(n.fullPath) != 0 && n.wildChild {
+							matchNum++
+						}
 						continue walk
 					}
 				}
 
+				// level 2 router not found,the current node needs to be equal to latestNode
+				if matchNum < 1 {
+					n = latestNode
+				}
+
 				// If there is no wildcard pattern, recommend a redirection
 				if !n.wildChild {
 					// Nothing found.
 					// We can recommend to redirect to the same URL without a
 					// trailing slash if a leaf exists for that path.
-					value.tsr = (path == "/" && n.handlers != nil)
+					value.tsr = path == "/" && n.handlers != nil
```

Comment:
my personal preference is to keep the parentheses, as they make it more clear you're assigning a boolean expression
```suggestion
					value.tsr = (path == "/" && n.handlers != nil)
```

Reply:
Thank you for your review, I see what you mean, but for a obsessive-compulsive disorder patients, in fact the parentheses is invalid and can be ignored, just had prompt effect (but not enough concise), if the judge conditions for more than two or more, or is too long, can adopt the way of line, I'm sorry, I want to insist on my opinion.

Reply:
#### There will also be differences in style
https://github.com/gin-gonic/gin/blob/9c27053243cb24ecc90d01c8ff379bd98fed9c8e/tree.go#L502

#### It makes sense under conditions like this

```go
var condition = (a &b) || 
                       (c && d)
```
https://github.com/gin-gonic/gin/blob/9c27053243cb24ecc90d01c8ff379bd98fed9c8e/tree.go#L558